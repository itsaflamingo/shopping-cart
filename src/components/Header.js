import React, {useEffect, useState} from 'react';
import shoppingCart from '../images/shopping-cart.png'
import Cart from './Cart'
import home from '../images/home.png'
import search from '../images/search.png'
import Search  from './Search';

export default function Header(props) {

    const [closeCart, setCloseCart] = useState(true);
    const [openSearch, setOpenSearch] = useState(false);
    const {total, newTotal, cart, navHome} = props;

  return (
      <div id='shop-header'>
        <h2>Shop</h2>
        <div id='buttons'>
        <div id='search' style={{
                backgroundImage: `url(${search})`,
                backgroundSize: 'contain',
                backgroundRepeat: 'no-repeat'
              }}></div>
        <div id='to-home' onClick={navHome} style={{
                backgroundImage: `url(${home})`,
                backgroundSize: 'contain',
                backgroundRepeat: 'no-repeat'
              }}></div>
        <div id='cart-icon' style={{
                backgroundImage: `url(${shoppingCart})`,
                backgroundSize: 'contain',
                backgroundRepeat: 'no-repeat'
              }} onClick={() => setCloseCart(!closeCart)}>
                <div id='cart-quantity'>{cart.length}</div>
            </div>
        </div>
        {!closeCart && <Cart items={cart} setCloseCart={setCloseCart} total={total} newTotal={newTotal} />}
        {openSearch && <Search />}
      </div>
  );
}