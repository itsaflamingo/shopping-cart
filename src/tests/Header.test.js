import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom"; 
import Header from '../components/Header'
import {BrowserRouter as Router} from 'react-router-dom';
import userEvent from "@testing-library/user-event";
import { act } from "react-dom/test-utils";

describe('Header component', () => {
    const navHome = jest.fn();
    const newTotal = jest.fn();
    const cart = [{id: '1'}];
    const total = 0;
    it('renders correct heading', () => {
        render(
            <Router>
                <Header navHome={navHome} cart={cart} newTotal={newTotal} total={total} />
            </Router>)
        expect(screen.getByRole('heading').textContent).toMatch(/Shop/i)
    })
    it('Goes to home page when home icon clicked', () => {
        const {asFragment} = render(
            <Router>
                <Header navHome={navHome} cart={cart} newTotal={newTotal} total={total}/>
            </Router>)
        const firstRender = asFragment();
        const button = screen.getByTestId('home-icon');
        act(() => {
            userEvent.click(button);
        })
        expect(firstRender).toMatchSnapshot();
    })
    it('Opens to cart when cart icon clicked', () => {
        const {asFragment} = render(
            <Router>
                <Header navHome={navHome} cart={cart} newTotal={newTotal} total={total}/>
            </Router>)
        const firstRender = asFragment();
        const button = screen.getByTestId('cart-icon');
        act(() => {
            userEvent.click(button);
        })
        expect(firstRender).toMatchSnapshot();
    })
    it('Cart number is 0 without items.', () => {
        const thisCart = [];
        render(
            <Router>
                <Header navHome={navHome} cart={thisCart} newTotal={newTotal} total={total}/>
            </Router>)
        
        const quantity = screen.queryByText('0');
        expect(quantity.innerHTML).toMatch(/0/i)
    })
    it('Cart display reflects number of items in cart array', () => {
        const thisCart = [{
                id: '1',
            }, {
                id: '2',
            }, {
                id: '3',
            }, {
                id: '4',
            }, {
                id: '5',
            }];
        render(
            <Router>
                <Header navHome={navHome} cart={thisCart} newTotal={newTotal} total={total}/>
            </Router>)
        
        const quantity = screen.queryByText('5');
        expect(quantity.innerHTML).toMatch(/5/i);
    })
} )
