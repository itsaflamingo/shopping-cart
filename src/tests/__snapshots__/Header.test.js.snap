// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Goes to home page when home icon clicked 1`] = `
<DocumentFragment>
  <div
    id="shop-header"
  >
    <h2>
      Fake Shop
    </h2>
    <div
      id="buttons"
    >
      <div
        id="search-container"
      >
        <div
          id="search"
          style="background-image: url(search.png); background-size: contain; background-repeat: no-repeat;"
        />
      </div>
      <div
        data-testid="home-icon"
        id="home-icon"
        style="background-image: url(home.png); background-size: contain; background-repeat: no-repeat;"
      />
      <div
        data-testid="cart-icon"
        id="cart-icon"
        style="background-image: url(shopping-cart.png); background-size: contain; background-repeat: no-repeat;"
      >
        <div
          id="cart-quantity"
        >
          1
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Header component Opens to cart when cart icon clicked 1`] = `
<DocumentFragment>
  <div
    id="shop-header"
  >
    <h2>
      Fake Shop
    </h2>
    <div
      id="buttons"
    >
      <div
        id="search-container"
      >
        <div
          id="search"
          style="background-image: url(search.png); background-size: contain; background-repeat: no-repeat;"
        />
      </div>
      <div
        data-testid="home-icon"
        id="home-icon"
        style="background-image: url(home.png); background-size: contain; background-repeat: no-repeat;"
      />
      <div
        data-testid="cart-icon"
        id="cart-icon"
        style="background-image: url(shopping-cart.png); background-size: contain; background-repeat: no-repeat;"
      >
        <div
          id="cart-quantity"
        >
          1
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
